### PR TITLE
k8s: add more test cases in k8s-env.bats

### LINKED ISSUE #4323: Kubernetes: add more test cases to k8s-env.bats
**Which feature do you think can be improved?**

The test cases in `integration/kubernetes/k8s-env.bats`

**How can it be improved?**

The kubernetes e2e suite provides a couple of [test cases](https://github.com/kubernetes/kubernetes/blob/release-1.22/test/e2e/common/node/downwardapi.go) for [downward API](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/). Examples of test cases:
 * [sig-node] Downward API should provide host IP as an env var
 * [sig-node] Downward API should provide container's limits.cpu/memory and  requests.cpu/memory as env vars 
 * [sig-node] Downward API should provide pod name, namespace and IP  address as env vars
 * [sig-node] Downward API should provide pod UID as env vars

We have some coverage in `integration/kubernetes/k8s-env.bats` but it can be increased to ensure it will export correctly values from different sources.

### PR DESCRIPTION
**Pull request by wainersm**
This increased the coverage of k8s-env.bats by adding cases where
different sources have their values loaded and exported in the container
environment.

Fixes #4323
Signed-off-by: Wainer dos Santos Moschetta <wainersm@redhat.com>

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @wainersm!

**Comment by wainersm** [comment]
The jenkins-ci-fedora-32-crio-k8s-e2e-minimal failure is related to https://github.com/kata-containers/tests/issues/4290 . I spent some time trying to figure out what causes that but I don't find the reason....

File: integration/kubernetes/k8s-env.bats [hunk 1]
```diff
@@ -24,6 +24,11 @@ setup() {
 	# Print environment variables
 	cmd="printenv"
 	kubectl exec $pod_name -- sh -c $cmd | grep "MY_POD_NAME=$pod_name"
+	kubectl exec $pod_name -- sh -c $cmd | grep "HOST_IP=\([0-9]\+\(\.\|$\)\)\{4\}"
+	# Requested 32Mi of memory
+	kubectl exec $pod_name -- sh -c $cmd | grep "MEMORY_REQUESTS=33554432"
```
Comment (jodh-intel) [inline integration/kubernetes/k8s-env.bats:29]: Might be clearer as:

```suggestion
	kubectl exec $pod_name -- sh -c $cmd | grep "MEMORY_REQUESTS=$((1024 * 1024 * 32))"
```

**Review by jodh-intel — APPROVED** [review]
Thanks @wainersm.

File: integration/kubernetes/k8s-env.bats (hunk 1 above)
Comment (wainersm) [reply integration/kubernetes/k8s-env.bats:29]: It is indeed clearer, I will change it.

**Comment by wainersm** [comment]
Updated this PR so that:
 * lines are kept less than 80 chars
 * `MEMORY_REQUESTS` is calculated, making it more clear to understand (@jodh-intel 's suggestion)

File: integration/kubernetes/k8s-env.bats (hunk 1 above)
Comment (wainersm) [reply integration/kubernetes/k8s-env.bats]: done

**Review by GabyCT — APPROVED** [review]

**Pull request merged**